vowpalwabbit/csoaa.cc

```diff
@@ -77,11 +80,24 @@ void predict_or_learn(csoaa& c, single_learner& base, example& ec)
   ec._reduction_features.template get<simple_label_reduction_features>().reset_to_default();
 
   bool dont_learn = DO_MULTIPREDICT && !is_learn;
+  if (probabilities)
+  {
+    base.multipredict(ec, 0, c.num_classes, c.pred, true);
+    for (uint32_t i = 0; i < c.num_classes; i++) ec.pred.scalars.push_back(c.pred[i].scalar);
+    float sum_prob = 0;
+    for (uint32_t i = 0; i < c.num_classes; i++)
+    {
+      ec.pred.scalars[i] = 1.f / (1.f + correctedExp(-c.pred[i].scalar));
```

Comment:
Would be nice to add a `normalize` function to [`VW::math`](https://vowpalwabbit.org/docs/vowpal_wabbit/cpp/latest/namespaceVW_1_1math.html) since this seems to be done a few times over the codebase

Reply:
Yea I agree, I'll add that in